lib/AtomicCache.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+namespace Amp\Cache;
+
+use Amp\Promise;
+use Amp\Sync\KeyedMutex;
+use Amp\Sync\Lock;
+use function Amp\call;
+
+final class AtomicCache implements Cache
```

Comment:
Is there any point of implementing `Cache`? If you're using `AtomicCache`, you're probably using the `AtomicCache` methods instead of the `Cache` methods.

Reply:
Good point. Removed the interface.